crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs

```diff
@@ -44,61 +44,60 @@ struct Actual {
     extra_stmts: Vec<Stmt>,
 }
 
-/// TODO: VisitMut
 #[fast_path(ShouldWork)]
-impl Fold for AsyncToGenerator {
-    noop_fold_type!();
+impl VisitMut for AsyncToGenerator {
+    noop_visit_mut_type!();
 
-    fn fold_module_items(&mut self, n: Vec<ModuleItem>) -> Vec<ModuleItem> {
-        self.fold_stmt_like(n)
+    fn visit_mut_module_items(&mut self, n: &mut Vec<ModuleItem>) {
+        self.visit_mut_stmt_like(n);
     }
 
-    fn fold_stmts(&mut self, n: Vec<Stmt>) -> Vec<Stmt> {
-        self.fold_stmt_like(n)
+    fn visit_mut_stmts(&mut self, n: &mut Vec<Stmt>) {
+        self.visit_mut_stmt_like(n);
     }
 }
 
 impl AsyncToGenerator {
-    fn fold_stmt_like<T>(&mut self, stmts: Vec<T>) -> Vec<T>
+    fn visit_mut_stmt_like<T>(&mut self, stmts: &mut Vec<T>)
     where
-        T: StmtLike + FoldWith<Actual>,
-        Vec<T>: FoldWith<Self>,
+        T: StmtLike + VisitMutWith<Actual>,
+        Vec<T>: VisitMutWith<Self>,
     {
-        let stmts = stmts.fold_children_with(self);
+        stmts.visit_mut_children_with(self);
 
-        let mut buf = Vec::with_capacity(stmts.len());
+        let mut stmts_updated = Vec::with_capacity(stmts.len());
 
-        for stmt in stmts {
+        for mut stmt in stmts.drain(..) {
             let mut actual = Actual {
                 in_object_prop: false,
                 in_prototype_assignment: false,
                 extra_stmts: vec![],
             };
-            let stmt = stmt.fold_with(&mut actual);
 
-            buf.extend(actual.extra_stmts.into_iter().map(T::from_stmt));
-            buf.push(stmt);
+            stmt.visit_mut_with(&mut actual);
+            stmts_updated.push(stmt);
+            stmts_updated.extend(actual.extra_stmts.into_iter().map(T::from_stmt));
         }
 
-        buf
+        stmts_updated.shrink_to_fit();
```

Comment:
This will be noop as `stmts_updated.len()` == `stmts.len() + sum(extra_stmts.len())`.